mypy/semanal_typeddict.py

```diff
@@ -294,6 +294,18 @@ def analyze_typeddict_classdef_fields(
                     )
                     if analyzed is None:
                         return None, [], set()  # Need to defer
+                    # TypedDictAnalyzer sets the AssignmentStmt type here, but
+                    # NamedTupleAnalyzer doesn't and instead has semanal.py set it
+                    # by calling analyze_class_body_common after.
+                    #
+                    # This is because unlike TypedDicts, NamedTuples support method
+                    # definitions. So, we must handle some of what analyze_class_body_common
+                    # does here -- including modifying `stmt.type`.
+                    #
+                    # TODO: Find some way of refactoring and partially unifying
+                    #       these two codepaths?
+                    if not has_placeholder(analyzed):
```

Comment:
I think https://github.com/python/mypy/pull/13516 might have been the root cause of the previous mypy_primer failures -- the previous version of this diff probably just never worked when we enabled recursive types, so it broke once I rebased pass the diff flipping the defaults.

I'm not really sure this is the correct fix though; it feels sort of ad-hoc.

I also need to figure out what's up with the bokeh error from mypy_primer -- I'm having a difficult time repro-ing those new failures locally.